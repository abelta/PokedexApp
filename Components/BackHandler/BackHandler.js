import { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { BackHandler } from 'react-native';
import { withRouter } from 'react-router-native';

class BackHandlerCustom extends PureComponent {
  constructor(props) {
    super(props);
    this.onHardwareBackPress = this.onHardwareBackPress.bind(this);
  }

  componentDidMount() {
    BackHandler.addEventListener('hardwareBackPress', this.onHardwareBackPress);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.onHardwareBackPress);
  }

  onHardwareBackPress() {
    const { history } = this.props;
    if (history.length > 0) {
      history.goBack();
      return true;
    }
    return false;
  }

  render() {
    return (null);
  }
}

BackHandlerCustom.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  history: PropTypes.object.isRequired,
};

export default withRouter(BackHandlerCustom);
