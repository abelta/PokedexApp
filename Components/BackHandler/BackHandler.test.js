import React from 'react';
import { MemoryRouter } from 'react-router-native';
import renderer from 'react-test-renderer';
import BackHandler from '.';

describe('BackHandler', () => {
  it('renders properly', () => {
    const tree = renderer.create(<MemoryRouter><BackHandler /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
