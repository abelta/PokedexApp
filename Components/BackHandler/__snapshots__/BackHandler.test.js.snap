// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackHandler renders properly 1`] = `null`;
